lib/modules/manager/cargo/locked-version.ts

```diff
@@ -0,0 +1,40 @@
+import { logger } from '../../../logger';
+import { readLocalFile } from '../../../util/fs';
+import { CargoLockSchema, CargoLockSchemaToml } from './schema';
+
+export async function extractLockFileVersions(
+  lockFilePath: string
+): Promise<Map<string, string[]>> {
+  const versionsByPackage = new Map<string, string[]>();
+  const content = await readLocalFile(lockFilePath, 'utf8');
+  if (!content) {
+    return versionsByPackage;
```

Comment:
return null on error and log debug message on calling code 
```suggestion
    return null;
```